cardano-api/src/Cardano/Api/IPC.hs

```diff
@@ -586,17 +592,26 @@ queryNodeLocalState :: forall mode result.
                     -> Maybe ChainPoint
                     -> QueryInMode mode result
                     -> IO (Either AcquiringFailure result)
-queryNodeLocalState connctInfo mpoint query = do
-    resultVar <- newEmptyTMVarIO
-    connectToLocalNode
-      connctInfo
+queryNodeLocalState connctInfo mpoint query =
+  runOopsInEither $ queryNodeLocalState_ connctInfo mpoint query
+
+queryNodeLocalState_ :: forall e mode result. ()
+  => e `CouldBe` AcquiringFailure
+  => LocalNodeConnectInfo mode
+  -> Maybe ChainPoint
+  -> QueryInMode mode result
+  -> ExceptT (Variant e) IO result
+queryNodeLocalState_ connectInfo mpoint query = do
+    resultVar <- liftIO newEmptyTMVarIO
+    liftIO $ connectToLocalNode
+      connectInfo
       LocalNodeClientProtocols {
         localChainSyncClient    = NoLocalChainSyncClient,
         localStateQueryClient   = Just (singleQuery mpoint resultVar),
         localTxSubmissionClient = Nothing,
         localTxMonitoringClient = Nothing
       }
-    first toAcquiringFailure <$> atomically (takeTMVar resultVar)
+    liftIO (atomically (takeTMVar resultVar)) & OO.onLeft (OO.throw . toAcquiringFailure)
```

Comment:
A note-worthy thing here is that aside from having to use `oops` functions to do error handling, the and the use of `liftIO`, which Haskell developers already ought to know, the implementation of the function is structurally the same.  `oops` does not force us to structure code in a particular way.

Moreover.  `oops` offers combinators that are already familiar.  `onLeft` already exists in `transformers-except` and we use it liberally.  `throw` is a familiar function in Haskell exception libraries.  `oops` tries to use familiar names for its combinators.